### PR TITLE
 Finishing #86

### PR DESCRIPTION
**Pull request by peteruithoven**
More changes to finish https://github.com/omnidan/redux-undo/issues/86.
- Updated unit tests 
- Remove initial state filter
- Added future filtering

### CODE REVIEW

**Comment by peteruithoven** [comment]
I'm ready for feedback. 
I've changed the title and message of the pull request because the changes went beyond just the unit tests.

File: test/index.spec.js [hunk 1]
```diff
@@ -153,19 +153,49 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
     describe('Actions', () => {
       it('should not record unwanted actions', () => {
         if (testConfig && testConfig.FOR_TEST_ONLY_excludedActions) {
-          // don't record this action in history
-          let decrementedState = mockUndoableReducer(mockInitialState, { type: testConfig.FOR_TEST_ONLY_excludedActions[0] })
-          expect(decrementedState.past).to.deep.equal(mockInitialState.past)
-          expect(decrementedState.future).to.deep.equal(mockInitialState.future)
+          const excludedAction = { type: testConfig.FOR_TEST_ONLY_excludedActions[0] }
+          const notFilteredReducer = undoable(countReducer, { ...testConfig, filter: null })
+          let expected = notFilteredReducer(mockInitialState, excludedAction)
```
Comment (omnidan) [inline test/index.spec.js]: small detail, but you can do:

```js
{
  ...notFilteredReducer(mockInitialState, excludedAction),
  wasFiltered: true
}
```

File: test/index.spec.js [hunk 2]
```diff
@@ -246,6 +276,23 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
           expect(undoInitialState.present).to.deep.equal(mockInitialState.present)
         }
       })
+
+      it('should undo to last not filtered state', () => {
```
Comment (omnidan) [inline test/index.spec.js]: nice :+1:

Comment (tiemevanveen) [inline test/index.spec.js]: Yeah great we have some tests for this.

File: test/index.spec.js [hunk 3]
```diff
@@ -312,8 +376,12 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
       })
 
       it('should increase the length of future if successful', () => {
-        if (incrementedState.past.length > jumpToPastIndex) {
-          expect(jumpToPastState.future.length).to.be.above(incrementedState.future.length)
+        // skip this test if steps are filtered out,
+        // because the action might have been was filtered it won't be added to the future
+        if (testConfig && !testConfig.FOR_TEST_ONLY_includeActions) {
+          if (incrementedState.past.length > jumpToPastIndex) {
+            expect(jumpToPastState.future.length).to.be.above(incrementedState.future.length)
```
Comment (tiemevanveen) [inline test/index.spec.js]: Check. Makes sense

**Pull request merged**